pkg/visibility/api/rest/pending_workload_CQ_test.go

```diff
@@ -0,0 +1,301 @@
+						Namespace: nsName,
+					},
+					LocalQueueName:         lqNameA,
+					Priority:               highPrio,
+					PositionInClusterQueue: 0,
+					PositionInLocalQueue:   0,
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "b",
+						Namespace: nsName,
+					},
+					LocalQueueName:         lqNameA,
+					Priority:               highPrio,
+					PositionInClusterQueue: 1,
+					PositionInLocalQueue:   1,
+				},
+			},
+		},
+		"query parameters offset set": {
+			clusterQueues: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue(cqNameA).Obj(),
+			},
+			queues: []*kueue.LocalQueue{
+				utiltesting.MakeLocalQueue(lqNameA, nsName).ClusterQueue(cqNameA).Obj(),
+			},
+			workloads: []*kueue.Workload{
+				utiltesting.MakeWorkload("a", nsName).Queue(lqNameA).Priority(highPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("b", nsName).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second)).Obj(),
+				utiltesting.MakeWorkload("c", nsName).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second * 2)).Obj(),
+			},
+			queryParams: &visibility.PendingWorkloadOptions{
+				Offset: 1,
+				Limit:  constants.DefaultPendingWorkloadsLimit,
+			},
+			wantPendingWorkloads: []visibility.PendingWorkload{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "b",
+						Namespace: nsName,
+					},
+					LocalQueueName:         lqNameA,
+					Priority:               highPrio,
+					PositionInClusterQueue: 1,
+					PositionInLocalQueue:   1,
+				},
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "c",
+						Namespace: nsName,
+					},
+					LocalQueueName:         lqNameA,
+					Priority:               highPrio,
+					PositionInClusterQueue: 2,
+					PositionInLocalQueue:   2,
+				},
+			},
+		},
+		"query parameters offset and limit set": {
+			clusterQueues: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue(cqNameA).Obj(),
+			},
+			queues: []*kueue.LocalQueue{
+				utiltesting.MakeLocalQueue(lqNameA, nsName).ClusterQueue(cqNameA).Obj(),
+			},
+			workloads: []*kueue.Workload{
+				utiltesting.MakeWorkload("a", nsName).Queue(lqNameA).Priority(highPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("b", nsName).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second)).Obj(),
+				utiltesting.MakeWorkload("c", nsName).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second * 2)).Obj(),
+			},
+			queryParams: &visibility.PendingWorkloadOptions{
+				Offset: 1,
+				Limit:  1,
+			},
+			wantPendingWorkloads: []visibility.PendingWorkload{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "b",
+						Namespace: nsName,
+					},
+					LocalQueueName:         lqNameA,
+					Priority:               highPrio,
+					PositionInClusterQueue: 1,
+					PositionInLocalQueue:   1,
+				},
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			manager := queue.NewManager(utiltesting.NewFakeClient(), nil)
+			ctx := context.Background()
```

Comment:
I think we should wrap the ctx with cancel and timeout, and defer cancel(), for example as here (and cleanup the manager): 
https://github.com/kubernetes-sigs/kueue/blob/1ecd79f748af2617111d0bfa1ca848f9e172f4d7/pkg/scheduler/scheduler_test.go#L1108-L1110